### PR TITLE
[SPARK-6138][CORE][minor] enhance the `toArray` method in `SizeTrackingVector`

### PR DESCRIPTION
**Pull request by cloud-fan**
Use array copy instead of `Iterator#toArray` to make it more efficient.

### CODE REVIEW

**Comment by srowen** [comment]
ok to test

**Comment by srowen** [comment]
Seems fine, though I am wondering if this is of a significance that needs a quick JIRA to track.

File: core/src/main/scala/org/apache/spark/util/collection/SizeTrackingVector.scala [hunk 1]
```diff
@@ -41,6 +41,6 @@ private[spark] class SizeTrackingVector[T: ClassTag]
    * Return a trimmed version of the underlying array.
    */
   def toArray: Array[T] = {
-    super.iterator.toArray
+    super.copyArrayWithLength(size)
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/util/collection/SizeTrackingVector.scala]: Hm, could this method also just be in the parent? it doesn't seem specific to the subclass even.

**Comment by cloud-fan** [comment]
hi @srowen , is it ready to go? or should I create a JIRA for it?

**Comment by srowen** [comment]
It's really minor, but if you wouldn't mind making a JIRA, that'd be a little more by-the-book here. I can resolve it rapidly. We usually skip JIRAs where the problem and solution are all but identical. That's almost true here.

**Pull request closed without merging**